src/PCGSolver_Diel.cc

```diff
@@ -149,29 +148,30 @@ void PCGSolver_Diel<T, T2>::preconSolve(
     if (bc_[0] != 1 || bc_[2] != 1 || bc_[2] != 1) gf_v.trade_boundaries();
 }
 
-template <class T, typename T2>
+template <class T, typename ScalarType>
 // Left Preconditioned CG
-bool PCGSolver_Diel<T, T2>::solve(
-    pb::GridFunc<T2>& gf_phi, pb::GridFunc<T2>& gf_rhs)
+bool PCGSolver_Diel<T, ScalarType>::solve(
+    pb::GridFunc<ScalarType>& gf_phi, pb::GridFunc<ScalarType>& gf_rhs)
 {
     if (!oper_.initialized())
     {
-        cout << "Error in PCGSolver_Diel<T>::solve: operator not initialized"
-             << endl;
+        std::cout
+            << "Error in PCGSolver_Diel<T>::solve: operator not initialized"
+            << std::endl;
         return 0.;
     }
 
     bool converged           = false;
     const pb::Grid& finegrid = gf_phi.grid();
 
     // initial data and residual - We assume a nonzero initial guess
-    pb::GridFunc<T2> lhs(finegrid, bc_[0], bc_[1], bc_[2]);
-    pb::GridFunc<T2> res(finegrid, bc_[0], bc_[1], bc_[2]);
+    pb::GridFunc<ScalarType> lhs(finegrid, bc_[0], bc_[1], bc_[2]);
+    pb::GridFunc<ScalarType> res(finegrid, bc_[0], bc_[1], bc_[2]);
     // scale initial guess with epsilon
     oper_.inv_transform(gf_phi);
     // compute initial residual
     oper_.apply(gf_phi, lhs);
-    pb::GridFunc<T2> rhs(gf_rhs);
+    pb::GridFunc<ScalarType> rhs(gf_rhs);
     oper_.transform(rhs);
     // Hartree units
     rhs *= (4. * M_PI);
```

Comment:
Should this be moved as well, like it was in the PCGSolver.cc code?

Reply:
It is not as straightforward. I think the whole class would need to be split into a "standard" solver and an equation specific part. TODO list